src/content/advanced/gitops/environments/index.md

```diff
@@ -0,0 +1,386 @@
+
+## Environments
+
+We are creating a new `base` type called environment. Inside there will be two folders.
+
+The `stages` folder is how we propose to group environment specifications. There is a good reason for this additional layer. You can use it for having multiple different clusters - like the dev, staging and production - but also to have multiple different regions where you want to spin these clusters up.
+
+The `regions` folder is how we propose to group specifications for configurations referred to different locations or regions in the different cloud providers or datacenters.
+
+__Note__: If you want to use multiple environment templates to create a single cluster that uses `App CR`s for deployments, for example, you would like to use `dev` layout to set app configuration and then use `east` from the `data-centers` to set the IP ranges, you might run into issues around merging configurations, as currently one configuration source (i.e. `ConfigMap` in `spec.config.configMap`) completely overrides the whole value of the same attribute coming from the other base. We're working to remove this limitation. This means for example that setting `machinePools` in several files will result on selecting only the full block of the file with higher priority.
+
+In order to avoid config collisions, we are specifying different config changes in different environment folders. In our case:
+
+- `stages` should refer to versions of the cluster and apps running, and different configurations associated to the stage.
+- `regions` should contain configurations referred to the locality, like networking or DNS zones.
+
+We're assuming that all the clusters using this environments pattern should in many regards look the same across all the environments. Still, each environment layer introduces some key differences, like the app version being deployed for `dev/staging/prod` environments or a specific IP range, availability zones, certificates or ingress config for regions like `eu-central/us-west`.
+
+To create an environment template, you need to make a  directory in `environments` that describes the best the
+differentiating factor for that kind of environment, then you should create a sub folder there for different possible values.
+
+For example, for multiple regions, we recommend putting region-specific configurations into
+`/bases/environments/regions` folder and under there create e.g. `eu_central`, and `us_west` folders.
+
+Once your environment templates are ready, you can use them to create new clusters by placing cluster definitions
+in `/management-clusters/MC_NAME/organizations/ORG_NAME/workload-clusters`
+
+### Stages
+
+Let's create 2 example type of clusters under `/bases/environments/stages`. Each of these contain a `example_cluster` folder with an example.
```

Comment:
```suggestion
Now, let's see how to put the explanation into practice. We create two cluster stage templates under `/bases/environments/stages`. 
```
IMO we don't need extra `example_cluster` folder, and it makes things harder. What if we discard it?